upgrade/class/Xoops/Upgrade/UpgradeControl.php

```
@@ -0,0 +1,327 @@
+
+        $upgrade_language = $xoopsConfig['language'] ?? null;
+        $upgrade_language = !empty($_COOKIE['xo_upgrade_lang']) ? $_COOKIE['xo_upgrade_lang'] : $upgrade_language;
+        $upgrade_language = \Xmf\Request::getString('lang', $upgrade_language);
+        $upgrade_language = $this->normalizeLanguage($upgrade_language);
+        xoops_setcookie('xo_upgrade_lang', $upgrade_language, 0, null, null);
+
+        $this->upgradeLanguage = $upgrade_language;
+        $this->loadLanguage('upgrade');
+
+        return $this->upgradeLanguage;
+    }
```

PHPMD (2.15.0)

[error] determineLanguage accesses the super-global variable $_COOKIE. (undefined)

(Superglobals)